crates/miden-objects/src/transaction/tx_header.rs

```diff
@@ -216,3 +232,79 @@ impl Deserializable for TransactionHeader {
         Ok(tx_header)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use miden_verifier::ExecutionProof;
+
+    use super::*;
+    use crate::asset::FungibleAsset;
+    use crate::transaction::ProvenTransactionBuilder;
+
+    #[test]
+    fn from_proven_transaction() {
```

Comment:
I'm not sure this tests is meaningful. I think what we should do instead is test this on real transactions. We can do this relatively easily by modifying the `prove_and_verify_transaction()` function since we get both proven and verified transactions there.